optimum/exporters/openvino/model_patcher.py

```diff
@@ -6456,3 +6456,83 @@ def __exit__(self, exc_type, exc_value, traceback):
             if layer.is_moe_layer:
                 layer.feed_forward.forward = layer.feed_forward._orig_forward
             layer.self_attn.forward = layer.self_attn._orig_forward
+
+
+class Zamba2ModelPatcher(DecoderModelPatcher):
+    def __init__(
+        self,
+        config: "OnnxConfig",
+        model: Union["PreTrainedModel", "TFPreTrainedModel"],
+        model_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        super().__init__(config, model, model_kwargs)
+
+    def __enter__(self):
+        self._model._orig_forward = self._model.forward
+
+        # the patch is needed to include KV-cache, Conv, and SSM states in the inputs and outputs.
+        def zamba2_forward(self, input_ids, attention_mask, position_ids, past_key_values):
+            from transformers.models.zamba2.modeling_zamba2 import Zamba2HybridDynamicCache
+
+            class CustomZamba2HybridDynamicCache(Zamba2HybridDynamicCache):
+                def __init__(self, config, batch_size: int, conv_states, ssm_states, key_cache, value_cache):
+                    """
+                    Custom Zamba2 cache object that holds conv, SSM, and KV states.
+
+                    Args:
+                        conv_states (torch.Tensor): Cached convolution states.
+                        ssm_states (torch.Tensor): Cached state-space model states.
+                        key_cache (torch.Tensor): Cached keys for attention.
+                        value_cache (torch.Tensor): Cached values for attention.
+                        config (PretrainedConfig): Model configuration object.
+                    """
+                    # Call parent constructor with all required arguments
+                    super().__init__(config=config, batch_size=batch_size)
+                    self.conv_states = conv_states
+                    self.ssm_states = ssm_states
+                    self.key_cache = key_cache
+                    self.value_cache = value_cache
```

Comment:
I don't think there's a need for a custom class, you could use the update method to populate the cache iteratively in the loop below.